modules/telestion-api/src/main/java/de/wuespace/telestion/api/config/Config.java

```diff
@@ -27,14 +28,16 @@
 	 * Selects the right configuration file.
 	 *
 	 * @param forcedConfig  this config will be applied if it is not null
-	 * @param defaultConfig this config contains the default values which replace absent values in the JSON config
+	 * @param defaultConfig this config contains the default values which replace absent values in the JSON config;
+	 *                      it must not be <code>null</code>!
 	 * @param jsonConfig    this is common config which represents the config class in json. If fields are not available
 	 *                      they will be set like the default values of the class when the default constructor is called.
-	 * @param type          the type of the config.
 	 * @param <T>           defines the class type of the configuration
 	 * @return the selected configuration
 	 */
-	public static <T> T get(T forcedConfig, T defaultConfig, JsonObject jsonConfig, Class<T> type) {
+	@SuppressWarnings("unchecked")
+	public static <T> T get(T forcedConfig, @NonNull T defaultConfig, JsonObject jsonConfig) {
```

Comment:
I guess it brakes all other code.

Why should we rely on unsafe code and revert a common pattern? 

You need to explain, how this impacts our code base in a good way. At the moment I don't see it.

Reply:
Yes it brakes stuff... However this allows for simpler code as `Class<T> type` is redundant for a `defaultConfig` which is not `null`.

As we usually call `Config.get(this.config, new Configuration(), config(), Configuration.class)` we could shorten this to `Config.get(this.config, new Configuration(), config())` as the defaultConfig is never `null` in that case.

Would you rather see an `unsafeGet()`?